### PR TITLE
HLP Refinements

### PR DESCRIPTION
**Pull request by san1526**
|Related issue|
|---|
|#13037|

## Description
- Added boolean parser
- Added 'to end of line' parser
- Added number parser (long and float)
- Changed ParseResult to be of type std::any
- Updated bench and UTs
- Added a function to configure hlp mappings
- Reworked the catalog to be simpler

### CODE REVIEW

**Review by palaciosjeremias — APPROVED** [review]
LGTM

**Review by juliancnn — APPROVED** [review]
LGTM

**Review by JcabreraC — APPROVED** [review]
LGTM !

File: src/engine/source/hlp/include/hlp/hlp.hpp [hunk 1]
```diff
@@ -1,13 +1,25 @@
 #ifndef _HLP_H
 #define _HLP_H
+#include <any>
```
Comment (nmkoremblum) [inline src/engine/source/hlp/include/hlp/hlp.hpp:3]:
```suggestion
#define _HLP_H

#include <any>
```

File: src/engine/source/hlp/src/hlp.cpp [hunk 2]
```diff
@@ -4,36 +4,87 @@
...
 #include <vector>
 
+#include <rapidjson/document.h>
+
 #include "hlpDetails.hpp"
 #include "logQLParser.hpp"
 #include "specificParsers.hpp"
 #include <hlp/hlp.hpp>
+#include <logging/logging.hpp>
 #include <profile/profile.hpp>
```
Comment (nmkoremblum) [inline src/engine/source/hlp/src/hlp.cpp:14]:
```suggestion
#include "hlpDetails.hpp"
#include "logQLParser.hpp"
#include "specificParsers.hpp"

#include <hlp/hlp.hpp>
#include <logging/logging.hpp>
#include <profile/profile.hpp>
#include <rapidjson/document.h>
```

File: src/engine/source/hlp/src/specificParsers.cpp [hunk 3]
```diff
@@ -18,6 +13,12 @@
 #include <rapidjson/stringbuffer.h>
 #include <rapidjson/writer.h>
 
+#include "date/date.h"
+#include "hlpDetails.hpp"
+#include "specificParsers.hpp"
+#include "tld.hpp"
+#include <profile/profile.hpp>
```
Comment (nmkoremblum) [inline src/engine/source/hlp/src/specificParsers.cpp:20]:
```suggestion
#include "date/date.h"
#include "hlpDetails.hpp"
#include "specificParsers.hpp"
#include "tld.hpp"

#include <profile/profile.hpp>
#include <rapidjson/stringbuffer.h>
#include <rapidjson/writer.h>
```

File: src/engine/test/source/hlp/hlp_test.cpp [hunk 4]
```diff
@@ -2,16 +2,33 @@
 
 #include "gtest/gtest.h"
```
Comment (nmkoremblum) [inline src/engine/test/source/hlp/hlp_test.cpp:3]:
```suggestion
#include <gtest/gtest.h>
```

**Review by nmkoremblum — CHANGES_REQUESTED** [review]

File: src/engine/source/builder/builders/stageParse.cpp [hunk 5]
```diff
@@ -9,20 +9,83 @@
...
-
 namespace builder::internals::builders
 {
+static bool
+any2Json(std::any const& anyVal, std::string const& path, json::Document* doc)
+{
+    auto& type = anyVal.type();
+    if (type == typeid(void))
+    {
+        doc->set(path, {nullptr, doc->getAllocator()});
```
Comment (juliancnn) [inline src/engine/source/builder/builders/stageParse.cpp:35]:
```suggestion
        json::Value val;
        val.SetNull();
        doc->set(path, val);
```

**Review by juliancnn — APPROVED** [review]

File: src/engine/source/builder/builders/stageParse.cpp (hunk 5 above)
Comment (LucioDonda) [reply src/engine/source/builder/builders/stageParse.cpp]: done!

File: src/engine/test/source/hlp/hlp_test.cpp (hunk 4 above)
Comment (LucioDonda) [reply src/engine/test/source/hlp/hlp_test.cpp]: done!

File: src/engine/source/hlp/src/specificParsers.cpp (hunk 3 above)
Comment (LucioDonda) [reply src/engine/source/hlp/src/specificParsers.cpp]: done!

File: src/engine/source/hlp/src/hlp.cpp (hunk 2 above)
Comment (LucioDonda) [reply src/engine/source/hlp/src/hlp.cpp]: done!

File: src/engine/source/hlp/include/hlp/hlp.hpp (hunk 1 above)
Comment (LucioDonda) [reply src/engine/source/hlp/include/hlp/hlp.hpp]: done!

**Comment by LucioDonda** [comment]
UT ok:
Screenshot from 2022-04-29 18-17-28

Benchmark:
Screenshot from 2022-04-29 18-15-57

Benchmark_tool for engine:
Screenshot from 2022-04-29 18-13-15

Pending modification of not found parser, this should return error notifying user

**Review by nmkoremblum — APPROVED** [review]
LGTM :)

**Review by juliancnn — APPROVED** [review]
LGTM! :rocket: 
For next iteration:
- Move error msg on build asset steps
- Add catalog test
- Remove cli tools HLP
- Add tracer

**Pull request merged**